[Enhancement] Show warning when ERC20 token allowance is too low

## Describe the issue in detail

We should address https://github.com/XLabs/portal-bridge-ui/issues/461 in Connect.

Repro:
 1. Attempt to send a token from an EVM network that you have not previously set a token allowance for (or "spending cap" as MetaMask calls it). You can view your current allowances at revoke.cash and entering in your address
 2. Initiate the transfer
 3. Note that your wallet will pop up with an approval to set a spending cap
 4. Set this spending cap to below what the default amount is (which will be prefilled with the amount you're trying to send from the first step)
 5. Once that transaction completes, you'll see "Error sending transfer, please try again", and the debug console will show an error like
 ```
 main.e527c64d.js:137 Error: call revert exception; VM Exception while processing transaction: reverted with reason string "insufficient allowance" [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="transferTokens(address,uint256,uint16,bytes32,uint256,uint32)", data="0x08c379a000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000016696e73756666696369656e7420616c6c6f77616e636500000000000000000000", errorArgs=["insufficient allowance"], errorName="Error", errorSignature="Error(string)", reason="insufficient allowance", code=CALL_EXCEPTION, version=abi/5.7.0)
 ```
<img width="1606" alt="Screenshot 2023-10-09 at 12 26 12 PM" src="https://github.com/wormhole-foundation/wormhole-connect/assets/689351/8a49dba2-be9b-4063-bbc0-aa77bfa4efd1">


## What is the expected/desired behavior

Error message should say: "Error due to insufficient token allowance, please fix and try again"
<img width="577" alt="Screenshot 2023-10-09 at 12 33 21 PM" src="https://github.com/wormhole-foundation/wormhole-connect/assets/689351/a9ecbcc9-375a-4323-a8f5-eba0ede3387c">

I'm also proposing that we change the default message to "Error with transfer, please try again". Sending + transfer seems redundant.

--- a/sdk/src/contexts/eth/context.ts
+++ b/sdk/src/contexts/eth/context.ts
@@ -34,6 +34,7 @@ import { arrayify } from 'ethers/lib/utils';
 import { ForeignAssetCache, chunkArray } from '../../utils';
 
 export const NO_VAA_FOUND = 'No message publish found in logs';
+export const INSUFFICIENT_ALLOWANCE = 'Insufficient token allowance';
 
 export class EthContext<
   T extends WormholeContext,
@@ -288,6 +289,15 @@ export class EthContext<
         overrides,
       );
       await tx.wait();
+      // Metamask allows users to set a different amount than specified above
+      // Check to make sure that the amount set was at least the requested amount
+      const nowApproved = await tokenImplementation.allowance(
+        senderAddress,
+        contractAddress,
+      );
+      if (nowApproved.lt(approveAmount)) {
+        throw new Error(INSUFFICIENT_ALLOWANCE);
+      }
     }
   }
 

--- a/wormhole-connect/src/views/Bridge/Send.tsx
+++ b/wormhole-connect/src/views/Bridge/Send.tsx
@@ -1,6 +1,9 @@
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { Context } from '@wormhole-foundation/wormhole-connect-sdk';
+import {
+  Context,
+  INSUFFICIENT_ALLOWANCE,
+} from '@wormhole-foundation/wormhole-connect-sdk';
 import { makeStyles } from 'tss-react/mui';
 
 import { CHAINS, TOKENS } from 'config';
@@ -138,9 +141,13 @@ function Send(props: { valid: boolean }) {
       dispatch(setRedeemRoute(route));
       dispatch(setAppRoute('redeem'));
       setSendError('');
-    } catch (e) {
+    } catch (e: any) {
       dispatch(setIsTransactionInProgress(false));
-      setSendError('Error sending transfer, please try again');
+      setSendError(
+        e?.message === INSUFFICIENT_ALLOWANCE
+          ? 'Error due to insufficient token allowance, please try again'
+          : 'Error with transfer, please try again',
+      );
       console.error(e);
     }
   }

